src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```
@@ -0,0 +1,998 @@
+			}
+
+			if (response == null) {
+				stats.put(String.format(BarcoE2Constant.GROUP_DEST_HASH_TAG_DEST_NAME_COLON_MEMBER, methodName, BarcoE2Constant.NONE, BarcoE2Constant.NONE), BarcoE2Constant.NONE);
+				return;
+			}
+			if (isSuperDest) {
+				// TODO routing control for super screen destination
+			} else {
+				SuperAuxDestination superAuxDestination = (SuperAuxDestination) jsonNodeToDTO(response, SuperAuxDestination.class);
+				populateSuperRouting(stats, controls, sourceList, methodName, superAuxDestination, false);
+			}
+		}
+	}
+
+	/**
+	 * Routing control: Get list of super screen/aux destination ids
+	 *
+	 * @param isSuperDest boolean true/false based on super destination/ super aux destination
+	 * @return List of ids
+	 */
+	private List<Integer> handleListSuperId(boolean isSuperDest) {
+		if (this.getListSuperScreenDestId() != null || this.getListSuperAuxDestId() != null) {
+			try {
+				List<String> resultList = new ArrayList<>();
+				if (isSuperDest) {
+					// Add string to List directly if length is 1
+					if (this.getListSuperScreenDestId().length() == 1) {
+						resultList.add(this.getListSuperScreenDestId());
+					} else {
+						// Split string array of IDs and populate to List
+						String[] listIds = this.getListSuperScreenDestId().split(BarcoE2Constant.COMMA);
+						Collections.addAll(resultList, listIds);
+					}
+				} else {
+					// Add string to List directly if length is 1
+					if (this.getListSuperAuxDestId().length() == 1) {
+						resultList.add(this.getListSuperAuxDestId());
+					} else {
+						// Split string array of IDs and populate to List
+						String[] listIds = this.getListSuperAuxDestId().split(BarcoE2Constant.COMMA);
+						Collections.addAll(resultList, listIds);
+					}
+				}
+				// stream the string List and convert to integer List
+				return resultList.stream().map(Integer::parseInt).collect(Collectors.toList());
+			} catch (Exception e) {
+				logger.error(e);
```

the same as above - is there a need to catch and rethrow an error? (Other than the message)
Also the original exception is lost (2nd parameter)

---

I changed it same as above.